README.md

```diff
@@ -11,19 +11,583 @@ JavaScript this Code-Along
+function can have properties, like `this`, and it can also have its own
+methods, in this case, `call`. Instead of invoking the `serve()`
+function directly, we're invoking the `call()` method of the `serve`
+function.
+
+The first argument for `call` is always the object that you wish to
+assign to `this` for the function. So by using `call(gc)`, we called the
+`serve` function, setting `this` to our `gc` sandwich object. That way,
+when we accessed `this.name` inside of the function, it knew the right
+sandwich for `this`.
+
+For a simple, no argument function like `serve()`, we can use `apply()`
+interchangeably with `call()`. The first argument for `apply()` is also
+always the object that we want to assign to `this` in the function, so
+`serve.apply(gc)` will work exactly the same as `serve.call(gc)`. Try it
+out!
+
+So both `call` and `apply` gives us a way to invoke a function and
+explicitly set `this` with the first argument. What is the difference?
+
+#### Passing Arguments With call() and apply()
+
+The only real difference between `call` and `apply` is the way you pass
+arguments to the target function.
+
+Let's modify our `serve` function to be a little friendlier:
+```js
+function serve(customer) {
+  console.log("Hey " + customer + ", here's your " + this.name + ", enjoy!");
+}
+
+var gc = new Sandwich("white", ["cheese"], "Grilled Cheese");
+var pbj = new Sandwich("wheat", ["peanut butter", "raspberry jam"],
+"Peanut Butter & Jelly");
+```
+
+Now, when we invoke `serve`, not only do we need to explicitly set
+`this`, but we need to pass in a value for `customer` as well.
+
+Using `call`, we simply pass the object for `this` as the first
+argument, and then any function arguments in order after that. So to use
+`serve` with `call`, let's do this:
+
+```js
+serve.call(gc, "Terry");
+serve.call(pbj, "Jesse");
+```
+
+Great! Now we see the name and the message! What happens if we don't
+pass that second argument?
+
+```js
+serve.call(gc);
+```
+
+We'll have an `undefined` value for `customer`. What if we skipped the
+`this` argument and just passed the customer value?
+
+```js
+serve.call("Terry");
+```
+
+Double `undefined`! Why? Think about what's happening here. We said that
+the first argument to `call()` or `apply()` is **always** the value for
+`this`, so inside our `serve` function, this is being set to the string
+literal `"Terry"`, and no value is being set to the `customer` variable.
+Inside of `serve`, when it tries to access `this.name`, it gets
+undefined, because the value of `this`, that string literal, doesn't
+have a `name` property.
+
+Okay, what about `apply`? Again, this works very similar to `call`,
+except that `apply` only takes two arguments: the value of `this`, and
+then an *array* of arguments to pass to the target function. So to use
+`apply` with our new serve object, we'll need to pass that customer
+value inside an array.
+
+```js
+serve.apply(gc, ["Terry"]);
```

Comment:
I really like these examples. Would it be possible to work with an example here that takes more than one argument? Maybe a `share()` function that takes two customers to really drive home the difference between `call()` and `apply()`?